backstory/src/main/java/com/google/sps/story/data/NLServiceClient.java

```diff
@@ -0,0 +1,185 @@
+
+    List<String> singleWords = new ArrayList<String>();
+
+    // remove multi-word inputs first and store as multiword nouns
+    for (String text : words) {
+      if (text.contains(" ")) {
+        if (map.containsKey(WordType.MULTIWORD_NOUN)) {
+          List<String> wordList = map.get(WordType.MULTIWORD_NOUN);
+          wordList.add(text);
+        } else {
+          ArrayList<String> wordList = new ArrayList<String>();
+          wordList.add(text);
+          map.put(WordType.MULTIWORD_NOUN, wordList);
+        }
+      } else {
+        singleWords.add(text);
+      }
+    }
+
+    for (String word : singleWords) {
+      Document doc = buildDocumentFromText(word);
+
+      AnalyzeSyntaxResponse response = client.analyzeSyntax(doc);
+
+      Token token = response.getTokens(0);
+      PartOfSpeech partOfSpeech = token.getPartOfSpeech();
+
+      WordType type = WordType.UNUSABLE;
+
+      switch (partOfSpeech.getTag()) {
+        case VERB:
+          if (isGerund(word)) {
+            type = WordType.GERUND;
+          }
+          break;
+        case NOUN:
+          if (isGerund(word)) {
+            type = WordType.GERUND;
+            break;
+          }
+
+          if (partOfSpeech.getProper() == Proper.PROPER) {
+            type = WordType.PROPER_NOUN;
+          } else {
+            type = WordType.NOUN;
+          }
+          break;
+
+        case ADJ:
+          type = WordType.ADJECTIVE;
+          break;
+      }
+
+      if (map.containsKey(type)) {
+        List<String> wordList = map.get(type);
+        wordList.add(word);
+      } else {
+        ArrayList<String> wordList = new ArrayList<String>();
+        wordList.add(word);
+        map.put(type, wordList);
+      }
+    }
+
+    return map;
+  }
+
+  /**
+   * Build and return a document from the given test.
+   *
+   * @param text the text to build a document from
+   * @return a Document that holds the current text
+   */
+  private Document buildDocumentFromText(String text) {
+    Document doc = Document.newBuilder().setContent(text).setType(Type.PLAIN_TEXT).build();
+
+    return doc;
+  }
+
+  /**
+   * Checks if a word is a gerund using heuristics (ending in "ing) and NL API
+   *
+   * @param word the word to check if it's a gerund
+   * @return true, if it's a gerund; false, otherwise
+   */
+  private boolean isGerund(String word) {
+    // first check if ends in "ing"
+    if (word.length() < 3) {
+      return false;
+    }
+
+    String ending = word.substring(word.length() - 3);
+    if (!ending.equals("ing")) {
+      return false;
+    }
+
+    // check if when paired with "is", it's identified as a verb
+    Document doc = buildDocumentFromText("is " + word);
+
+    AnalyzeSyntaxResponse response = client.analyzeSyntax(doc);
+
+    // get the second token (the potential gerund)
+    Token token = response.getTokens(1);
+
+    if (token.getPartOfSpeech().getTag() == Tag.VERB) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
This could be condensed to:
return token.getPartOfSpeech().getTag() == Tag.VERB;

Reply:
I agree. I'll change this.

Reply:
Done